server/edd/load/parsers/core.py

```diff
@@ -641,6 +645,127 @@ def _raw_cell_value(self, cell):
         return val
 
 
+class MultiSheetExcelParserMixin:
+    def parse(self, file):
+        """
+        Parses the input as a Pandas dataframe and then
+        converts it into an Excel workbook.
+
+        :param file: a file-like object
+        :return: the ParseResult read from file, otherwise None
+        :raises OSError: if the file can't be opened
+        :raises EDDImportError: if the file format or content is bad
+        """
+
+        wb = pd.read_excel(file, sheet_name=None)
+
+        # for each worksheet in the workbook
+        parsed_result = pd.DataFrame()
+        mapper = MeasurementMapper()
+
+        for name, sheet in wb.items():
+            # for every two columns in the worksheet
+            # corresponding to each measurement type in the sheet
+            for i in range(0, int(sheet.shape[1]), 2):
+                two_cols = sheet[sheet.columns[i : i + 2]]
+
+                # dropping all rows with nan values in the worksheet
+                two_cols = two_cols.dropna()
+                if not two_cols.dropna().empty:
+                    # decimate the data
+                    two_cols = two_cols.iloc[::10, :]
```

Comment:
This does not belong in the parser stage. The parser should read *all* of the data, then pass it along. The code that converts parsed and resolved data into database records should make the determination of which items to drop, and whether to do anything more complicated than a blind, one-in-ten sampling.